utils/helpers.js

```diff
@@ -0,0 +1,167 @@
+        let res = await axios({
+          method: 'post',
+          url: `${remakeServiceHost}/service/login`, 
+          data: {
+            email: loginAnswers.email,
+            password: loginAnswers.password,
+          }
+        });
+        remakeCliConfig.set('user.email', loginAnswers.email);
+        remakeCliConfig.set('user.authToken', res.data.token);
+        spinner.succeed('You are successfuly logged in.');
+      } catch (err) {
+        spinner.fail('Could not log you in. Please try again.');
+        process.exit();
+      }
+    } else {
+      try{
+        spinner = ora('Creating your account').start();
+        let res = await axios({
+          method: 'post',
+          url: `${remakeServiceHost}/service/signup`, 
+          data: {
+            email: loginAnswers.email,
+            password: loginAnswers.password,
+          }
+        });
+        remakeCliConfig.set('user.email', loginAnswers.email);
+        remakeCliConfig.set('user.authToken', res.data.token);
+        spinner.succeed('Created your account and logged you in.');
+      } catch (err) {
+        spinner.fail('Could not create your account. Please try again.');
+        process.exit();
+      }
+    }
+  }
+}
+
+const checkSubdomain = async (subdomain) => {
+  try {
+    const availabilityRes = await axios({
+      method: 'get',
+      url: `${remakeServiceHost}/service/subdomain/check`, 
+      headers: {
+        'Authorization': `Bearer ${remakeCliConfig.get('user.authToken')}`
+      },
+      params: {
+        subdomain,
+      }
+    });
+    if (availabilityRes.status === 200) return true;
+    else return false;
+  } catch (err) {
+    return false;
+  }
+}
+
+const registerSubdomain = async (subdomain) => {
+  try {
+    const domainRegistered = await axios({
+      method: 'post',
+      url: `${remakeServiceHost}/service/subdomain/register`, 
+      headers: {
+        'Authorization': `Bearer ${remakeCliConfig.get('user.authToken')}`
+      },
+      data: {
+        subdomain,
+      }
+    });
+    if (domainRegistered.status === 200) return { success: true };
+  } catch (err) {
+    return { success: false, message: err.response.data.message };
+  }
+}
+
+const createDeploymentZip = (projectName) => {
+  const spinner = ora('Archiving files for upload.').start();
+  return new Promise((resolve, reject) => {
+    const cwd = process.cwd();
+    const output = fs.createWriteStream(path.join(cwd, `deployment-${projectName}.zip`), { encoding: 'base64' });
+    const archive = archiver('zip', { zlib: { level: 9 } });
+
+    output.on('warning', (err) => {
+      spinner.fail();
+      reject(err);
+    });
+
+    output.on('error', (err) => {
+      spinner.fail();
+      reject(err);
+    });
+
+    output.on('close', () => {
+      spinner.succeed('Done archiving: ' + archive.pointer() + ' bytes.');
+      resolve();
+    })
+
+    archive.pipe(output);
+    archive.glob('app/[a-z]*/**/*');
```

Comment:
just curious, why the `[a-z]` check — why not everything?

Reply:
I think it's because I had some .directory in the project I was working with. I checked again and that .directory is gone.

Reply:
I don't see any harm in zipping the whole thing up. It doesn't widen the attack surface area and people might want to include weirdly named files.